src/test/ui/span/issue-39018.stderr

```diff
@@ -35,6 +35,145 @@ help: `to_owned()` can be used to create an owned `String` from a string referen
 LL |     let x = "Hello ".to_owned() + &"World!".to_owned();
    |             ^^^^^^^^^^^^^^^^^^^   ^^^^^^^^^^^^^^^^^^^^
 
-error: aborting due to 3 previous errors
+error[E0369]: binary operation `+` cannot be applied to type `&std::string::String`
+  --> $DIR/issue-39018.rs:26:16
+   |
+LL |     let _ = &a + &b;
+   |             -- ^ -- &std::string::String
+   |             |  |
+   |             |  `+` can't be used to concatenate two `&str` strings
+   |             &std::string::String
+help: String concatenation appends the string on the right to the string on the left and may require reallocation. This requires ownership of the string on the left
```

Comment:
I feel a bit of information overload here; there's too much information packed in too little space. I'd consider giving things more "air" by e.g. moving `help:` down a line.

Reply:
That is handled elsewhere in the emitter code. I am ambivalent on wether making this case take more vertical by adding a `\n|` would actually improve things in general or not. I feel we should be fairly conservative when it comes to vertical space, and with colors it doesn't look that bad:

<img width="695" alt="" src="https://user-images.githubusercontent.com/1606434/57900374-6d01ed80-7815-11e9-9360-44eb7b567e9d.png">

Reply:
Hmm, it would have been nice to have a method on `DiagnosticBuilder` that forces a newline... but since it happens elsewhere... bummer! and let's move on.

I do agree colors make it less of a problem, but that still makes me somewhat dizzy.

Reply:
Yeah, these look much better when the text is much shorter than this. What we could do is make the help text short and only say what it should do (fix the types of the binop), and have a separate note talk about this, which would introduce a newline and look like this

```

error[E0369]: binary operation `+` cannot be applied to type `&std::string::String`
  --> $DIR/issue-39018.rs:33:15
   |
LL |     let _ = e + &d;
   |             - ^ -- &&str
   |             | |
   |             | `+` cannot be used to concatenate two `&str` strings
   |             &std::string::String
   = note: string concatenation appends the string on the right to the string on the left and may require reallocation, and this requires ownership of the string on the left
help: turn the string in the lift into a `String`
   |
LL |     let _ = e.to_owned() + &d;
   |             ^^^^^^^^^^^^

Reply:
That seems better, yeah.